### DISCUSSION TITLE
Shipping binaries with homebrew

### DISCUSSION THREAD

**Discussion by danibachar** [Homebrew/discussions#4439] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.14-74-g71de574
ORIGIN: https://github.com/Homebrew/brew
HEAD: 71de57428568292bd3f5aa52250dd4eb39c3453e
Last commit: 2 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 0cc2d34f5a7ad1b4716a764b08eb616cef7b2763
Core tap last commit: 42 minutes ago
Core tap branch: tuist
Core tap JSON: 19 Apr 17:20 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode-14.3.0.app/Contents/Developer/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: N/A
Xcode: 14.3 => /Applications/Xcode-14.3.0.app/Contents/Developer
Rosetta 2: false
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

I have read through the threads regarding universal binaries with homebrew [here](https://github.com/orgs/Homebrew/discussions/412#discussioncomment-396027) and [here](https://github.com/Homebrew/brew/issues/10307).

There is a universal binary that is built outside of homebrew as part of some CI, and stored in github release zip.
All I want to do is ship this binary with home brew instead of download it (for example via curl) and copy it to some system path (for example `/usr/local/bin`).

I wrote a Formula, installed and tested.

Note the formula is super simple, as its just unzip the uploaded binary
```
class Tuist < Formula
  desc "Basic fomula that install tuist cli tool, installs latest version"
  homepage "https://tuist.io/"
  url "https://github.com/tuist/tuist/releases/download/3.18.0/tuistenv.zip"
  sha256 "601543c96c50aaf1850951b1ad5e04ae73ccf3ab1d862545f673df311c076ee1"
  license "MIT"

  uses_from_macos "ruby"

  def install
    # rename Tuist, here the zip file is already downloaded and unzipped
    mv("tuistenv", "tuist")
    # Support CLI running
    chmod("+x", "tuist")
    # configue CLI tool
    bin.install "tuist"
  end

  test do
    system "#{bin}/tuist", "--help"
  end
end

```

But when I run
`brew audit --new tuist` and `brew audit --strict --new tuist`
I get the following error:
```
Unexpected universal binaries were found.
    The offending files are:
      /opt/homebrew/Cellar/tuist/3.18.0/bin/tuist
```

I understand why homebrew will not support building universal library, and for some it might look weird to support a simple shipping of the binary using homebrew, but I like homebrew and am using it as a universal package manager on my mac - so shipping custom binaries can be useful...

Can homebrew support this usecase?

**Comment by ZhongRuoyu** [comment, marked as answer]
If this is hosted in a tap, one can bypass the audit by adding the formula name to [`audit_exceptions/universal_binary_allowlist.json`](https://github.com/Homebrew/homebrew-core/blob/master/audit_exceptions/universal_binary_allowlist.json).

Do note that `Homebrew/core` does not accept formulae that ship pre-built binaries (see ["We don’t like binary formulae"](https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae)).

**Reply by danibachar** [reply]
For future reader. I overviewed the documentation more throughly and understood that Binaries should be published using Cask repo. Merged it there. On caveat is that cask doesn’t support versioning. Which Im ok with and using the latest version

Thanks for the quick responses and info
